includes/discovery/discovery-protocols.inc.php

```diff
@@ -140,60 +140,64 @@
         $remote_device_id = find_device_id($lldp['lldpRemSysName']);
 
         if (! $remote_device_id &&
-            \LibreNMS\Util\Validate::hostname($lldp['lldpRemSysName']) &&
-            ! can_skip_discovery($lldp['lldpRemSysName'], $lldp['lldpRemSysDesc'] &&
-            Config::get('autodiscovery.xdp') === true)
-        ) {
+                \LibreNMS\Util\Validate::hostname($lldp['lldpRemSysName']) &&
+                ! can_skip_discovery($lldp['lldpRemSysName'], $lldp['lldpRemSysDesc'] &&
+                    Config::get('autodiscovery.xdp') === true)
+           ) {
             $remote_device_id = discover_new_device($lldp['lldpRemSysName'], $device, 'LLDP', $interface);
         }
 
         if ($interface['port_id'] && $lldp['lldpRemSysName'] && $lldp['lldpRemPortId']) {
             $remote_port_id = find_port_id($lldp['lldpRemPortDesc'], $lldp['lldpRemPortId'], $remote_device_id);
             discover_link(
-                $interface['port_id'],
-                'lldp',
-                $remote_port_id,
-                $lldp['lldpRemSysName'],
-                $lldp['lldpRemPortId'],
-                null,
-                $lldp['lldpRemSysDesc'],
-                $device['device_id'],
-                $remote_device_id
-            );
+                    $interface['port_id'],
+                    'lldp',
+                    $remote_port_id,
+                    $lldp['lldpRemSysName'],
+                    $lldp['lldpRemPortId'],
+                    null,
+                    $lldp['lldpRemSysDesc'],
+                    $device['device_id'],
+                    $remote_device_id
+                    );
         }
     }//end foreach
     echo PHP_EOL;
 } elseif ($device['os'] == 'timos') {
     echo ' TIMETRA-LLDP-MIB: ';
-    $lldp_array = SnmpQuery::hideMib()->walk('TIMETRA-LLDP-MIB::tmnxLldpRemoteSystemsData')->table();
-    foreach ($lldp_array as $key => $lldp) {
-        $ifIndex = key($lldp['tmnxLldpRemPortId']);
-        $MacIndex = key($lldp['tmnxLldpRemPortId'][$ifIndex]);
-        $RemIndex = key($lldp['tmnxLldpRemPortId'][$ifIndex][$MacIndex]);
-        $interface = get_port_by_ifIndex($device['device_id'], $ifIndex);
-        $remote_device_id = find_device_id($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex]);
-
-        if (! $remote_device_id &&
-            \LibreNMS\Util\Validate::hostname($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex]) &&
-            ! can_skip_discovery($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex], $lldp['tmnxLldpRemSysDesc'][$ifIndex][$MacIndex][$RemIndex]) &&
-            Config::get('autodiscovery.xdp') === true
-        ) {
-            $remote_device_id = discover_new_device($lldp['tmnxLldpRemSysName'][$ifIndex][$MacIndex][$RemIndex], $device, 'LLDP', $interface);
-        }
+    $lldp_array = SnmpQuery::hideMib()->walk('TIMETRA-LLDP-MIB::tmnxLldpRemoteSystemsData')->table(4);
```

Comment:
Use of table(4) to get a correctly structured associative array. Makes everything simpler afterward.